test/integration/replicaset/replicaset_test.go

```diff
@@ -272,246 +280,595 @@ func TestAdoption(t *testing.T) {
 	}
 }
 
-func createRSsPods(t *testing.T, clientSet clientset.Interface, rss []*v1beta1.ReplicaSet, pods []*v1.Pod, ns string) {
+func createRSsPods(t *testing.T, clientSet clientset.Interface, rss []*v1beta1.ReplicaSet, pods []*v1.Pod, ns string) ([]*v1beta1.ReplicaSet, []*v1.Pod) {
 	rsClient := clientSet.Extensions().ReplicaSets(ns)
 	podClient := clientSet.Core().Pods(ns)
+	var createdRSs []*v1beta1.ReplicaSet
+	var createdPods []*v1.Pod
 	for _, rs := range rss {
-		if _, err := rsClient.Create(rs); err != nil {
+		createdRS, err := rsClient.Create(rs)
+		if err != nil {
 			t.Fatalf("Failed to create replica set %s: %v", rs.Name, err)
 		}
+		createdRSs = append(createdRSs, createdRS)
 	}
 	for _, pod := range pods {
-		if _, err := podClient.Create(pod); err != nil {
+		createdPod, err := podClient.Create(pod)
+		if err != nil {
 			t.Fatalf("Failed to create pod %s: %v", pod.Name, err)
 		}
+		createdPods = append(createdPods, createdPod)
 	}
+
+	return createdRSs, createdPods
 }
 
+// Verify .Status.Replicas is equal to .Spec.Replicas
 func waitRSStable(t *testing.T, clientSet clientset.Interface, rs *v1beta1.ReplicaSet, ns string) {
 	rsClient := clientSet.Extensions().ReplicaSets(ns)
-	if err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
-		updatedRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
-		if updatedRS.Status.Replicas != *rs.Spec.Replicas {
-			return false, nil
-		} else {
+		if newRS.Status.Replicas == *newRS.Spec.Replicas {
 			return true, nil
 		}
+		return false, nil
 	}); err != nil {
-		t.Fatal(err)
+		t.Fatalf("Failed to verify .Status.Replicas is equal to .Spec.Replicas for rs %s: %v", rs.Name, err)
 	}
 }
 
-func TestUpdateSelectorToAdopt(t *testing.T) {
-	// We have pod1, pod2 and rs. rs.spec.replicas=1. At first rs.Selector
-	// matches pod1 only; change the selector to match pod2 as well. Verify
-	// there is only one pod left.
-	s, closeFn, rm, informers, clientSet := rmSetup(t)
+// selectors are IMMUTABLE for all API versions except extensions/v1beta1
+func TestRSSelectorImmutability(t *testing.T) {
```

Comment:
Git diff can be confusing. To clarify, I didn't change any codes for `TestRSSelectorImmutability`.

Reply:
Why are you testing selector immutability? We verify immutability in the validation testing. Why is this a valuable test?

Reply:
As discussed by person, this test was added on the selector immutability PR. There is no any change to this test in this PR (git diff can be confusing).